### PR TITLE
Add eslintrc rules consistent w/ codebase + Travis

### PR DESCRIPTION
**Pull request by cmc333333**
Rather than go without a linter (our current strategy), let's draw a line in
the sand and prevent future issues by adding a linter that matches our current
code base. We're breaking some pretty important rules (e.g. no-undef), so we
should revisit these to fix things ASAP.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,7 +2,34 @@
     "extends": "airbnb",
     "rules": {
         // Override any settings from the airbnb configuration
-        "indent": [2, 4],
-        "comma-dangle": 0
+        // "indent": [2, 4],
```
Comment (nmonga91) [inline .eslintrc]: is this override being removed because the default is sufficient?

File: .eslintrc [hunk 2]
```diff
@@ -2,7 +2,34 @@
...
+        "comma-dangle": 0,
+        // To fix
+        "brace-style": 0,
+        "camelcase": 0,
```
Comment (nmonga91) [inline .eslintrc]: Does `0` indicate "don't override and use the default" ?

File: src/js/helpers/reviewHelper.js [hunk 3]
```diff
@@ -172,7 +172,7 @@ const getFileStates = (status) => {
 		else {
 			let count = 0;
 			item.error_data.forEach((error) => {
-				count += parseInt(error.occurrences);
+				count += parseInt(error.occurrences, 10);
```
Comment (nmonga91) [inline src/js/helpers/reviewHelper.js]: The `10` here is to ensure the result is in base 10. But that should be the default (provided that the number in question doesn't start with a 0), no? 
Is it possible for `error.occurrences` to start with a 0?

File: .eslintrc (hunk 1 above)
Comment (cmc333333) [inline .eslintrc]: Nah, I commented it out (and turned it off later) because our current code doesn't match this style.

File: .eslintrc (hunk 2 above)
Comment (cmc333333) [inline .eslintrc]: `0` is the same as `off`, it means "ignore this rule". So we're turning off rules that were turned on in the airbnb style guide.

File: src/js/helpers/reviewHelper.js (hunk 3 above)
Comment (cmc333333) [inline src/js/helpers/reviewHelper.js]: Good question! Adding the `base` parameter to `parseInt` is just a sanity check. We almost never want to parse an integer in a non-10 base (and when we do, we should be explicit that that's the intent), so one of the [linting rules](http://eslint.org/docs/rules/radix) checks for this. I think that's a reasonable standard, but would be happy to revert if there's contention.

Comment (nmonga91) [inline src/js/helpers/reviewHelper.js]: I think we should keep it if the linting rules are checking for it. Being explicit never hurt so I'm definitely ok with leaving it in.

**Comment by nmonga91** [comment]
Looks good, merging

**Pull request merged**